### PR TITLE
Migration of //kccapi:spec

### PR DESCRIPTION
**Pull request by alfredomusumeci**
The first issue has been solved (https://github.com/goldmansachs/crdtools/issues/4). In particular, SpecExtractor has been refactored and comments have been added. The relative WORKSPACE and BUILD files have also been updated to reflect these changes. It is now possible to run bazel build //:spec in the root folder as requested.
Tests for the SpecExtractor have not been added as the only ones I could come up with would only check if the outputs files were indeed generated - my reasoning for not adding them is that the relative path of the output results will change depending on where the .jar file (and corresponding bazel command) is executed; therefore they are easy to break and not really meaningful in terms of the task's success. However, I am open to a second opinion!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,3 +25,12 @@ choose "Manage Plugin Repositories". Add https://plugins.jetbrains.com/plugins/b
 the list of repositories. After this is done, the bazel plugin compatible with recent IntelliJ
 versions should be available by clicking "Marketplace" in the Plugins and searching for Bazel.
 This has been tested with IntelliJ Idea Community 2022.1.2.
+
+## Credits
+Part of this code has been previously developed by the CTL team at [GS](https://gs.com). This
```
Comment (nresare) [inline README.md:30]: This would need coordination with PR people. I think we should go with something short and sweet. "The code that this project is based on was written by Jonathan Perry at Goldman Sachs"

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,5 @@ bazel-crdtools
 bazel-out
 bazel-bin
 bazel-testlogs
+.idea
+*.iml
```
Comment (nresare) [inline .gitignore]: No newline at end of file

File: src/test/java/com/gs/crdtools/SpecExtractorTest.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.gs.crdtools;
+
+public class SpecExtractorTest {
```
Comment (nresare) [inline src/test/java/com/gs/crdtools/SpecExtractorTest.java:3]: Let's only check in code that actually does something

File: WORKSPACE [hunk 4]
```diff
@@ -1,47 +1,122 @@
...
+
+http_archive(
+    name = "com_github_bazelbuild_buildtools",
+    sha256 = "ae34c344514e08c23e90da0e2d6cb700fcd28e80c02e23e4d5715dddcb42f7b3",
+    strip_prefix = "buildtools-4.2.2",
+    urls = [
+        "https://github.com/bazelbuild/buildtools/archive/refs/tags/4.2.2.tar.gz",
+    ],
+)
+
```
[the remark this answers is not in the document]
Comment (nresare) [reply WORKSPACE:59]: Wrong, line 2 is needed

File: BUILD [hunk 5]
```diff
@@ -1 +1,15 @@
 # This file exists so that the intellij Bazel plugin has something to bootstrap the project from
+
+load("@com_github_bazelbuild_buildtools//buildifier:def.bzl", "buildifier")
+
+buildifier(
```
Comment (nresare) [inline BUILD:5]: If we want to add this, let's do it in a separate MR

**Comment by nresare** [comment]
Could you merge this into a single commit?

File: README.md (hunk 1 above)
Comment (alfredomusumeci) [reply README.md]: Updated README file

File: .gitignore (hunk 2 above)
Comment (alfredomusumeci) [reply .gitignore]: Added EOL

File: WORKSPACE (hunk 4 above)
[the remark this answers is not in the document]
Comment (alfredomusumeci) [reply WORKSPACE]: Deleted unnecessary code

File: src/test/java/com/gs/crdtools/SpecExtractorTest.java (hunk 3 above)
Comment (alfredomusumeci) [reply src/test/java/com/gs/crdtools/SpecExtractorTest.java]: This must've escaped my eyes while committing, apologies for that

File: BUILD (hunk 5 above)
Comment (alfredomusumeci) [reply BUILD]: Ok!

File: src/main/java/com/gs/utils/VavrHelpers.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
+package com.gs.utils;
```
[the remark this answers is not in the document]
Comment (alfredomusumeci) [reply src/main/java/com/gs/utils/VavrHelpers.java]: Refactored the project structure
